Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1540,6 +1540,13 @@ private int GetPointerStateIndexFor(InputControl control)
                 m_CurrentPointerId = pointerId;
                 m_CurrentPointerType = pointerType;
 
+                if (point != null && point.action != null)
+                {
+                    var state = m_PointerStates[m_CurrentPointerIndex];
+                    state.screenPosition = point.action.ReadValue<Vector2>();
```

Comment:
This won't work reliably due to the nature of pass-through actions. The read will go to the last active control, not necessarily to the one that correlates with the control we're looking for here.